### ISSUE TITLE
make lib-ext fails patching cmdliner on alpine / busybox

### ISSUE THREAD

**Issue by talex5** [ocaml/opam#4021]
```
# git describe
2.0.0-531-g2ab2a6e4

# make lib-ext
make -j -C src_ext lib-ext
make[1]: Entering directory '/tmp/opam/src_ext'
mkdir -p tmp-cmdliner
cd tmp-cmdliner && bunzip2 -c ../cmdliner.tbz | tar xf -
rm -f seq/src/seq.ml seq/src/seq.mli
rm -rf cmdliner
patching file src/cmdliner_docgen.ml
patching file _tags
Hunk 1 FAILED 1/1.
-true : bin_annot, safe_string
+true : bin_annot, safe_string, package(bytes), package(result)
 <src> : include
-<test> : include
patching file pkg/META
[...]
make[1]: *** [Makefile:173: cmdliner.stamp] Error 1
make[1]: Leaving directory '/tmp/opam/src_ext'
make: *** [Makefile:60: lib-ext] Error 2
```

Here's a `Dockerfile` that shows the problem:

```
FROM alpine:3.10
RUN apk update && apk upgrade
RUN apk add build-base bzip2 git tar curl ca-certificates openssl
RUN git clone git://github.com/ocaml/opam /tmp/opam
WORKDIR /tmp/opam
RUN make lib-ext
```

Running `make cold` instead of `make lib-ext` fails in the same way (but takes much longer).

**Comment by rjbou** [comment]
Hmm... it comes from `busybox`' `patch `...
```
$ tar xvf cmdliner.tbz && cd cmdliner-1.0.4/
cmdliner-1.0.4/B0.ml
[...]
$ patch -p1 < ../patches/cmdliner.common/0001-Restore-4.02.3-support.patch 
patching file _tags
patching file pkg/META
Hunk #1 succeeded at 1 with fuzz 2.
patching file src/cmdliner.ml
Hunk #1 succeeded at 4 with fuzz 1.
patching file src/cmdliner.mli
patching file src/cmdliner_arg.ml
Hunk #1 succeeded at 4 with fuzz 1.
patching file src/cmdliner_arg.mli
Hunk #1 succeeded at 4 with fuzz 1.
patching file src/cmdliner_base.ml
patching file src/cmdliner_cline.ml
Hunk #1 succeeded at 4 with fuzz 1.
patching file src/cmdliner_cline.mli
Hunk #1 succeeded at 4 with fuzz 1.
patching file src/cmdliner_docgen.ml
patching file src/cmdliner_manpage.ml
Hunk #1 succeeded at 4 with fuzz 1.
patching file src/cmdliner_term.ml
Hunk #1 succeeded at 4 with fuzz 1.
patching file src/cmdliner_term.mli
Hunk #1 succeeded at 4 with fuzz 1.
patching file src/dune
$ cd .. && rm -rf cmdliner-1.0.4 && tar xvf cmdliner.tbz && cd cmdliner-1.0.4/
[...]
$ busybox patch -p1 < ../patches/cmdliner.common/0001-Restore-4.02.3-support.patch 
patching file _tags
Hunk 1 FAILED 1/1.
-true : bin_annot, safe_string
+true : bin_annot, safe_string, package(bytes), package(result)
 <src> : include
-<test> : include
patching file pkg/META
[...]
```

**Comment by dbuenzli** [comment]
My offer to [participate to global warming](https://github.com/ocaml/opam/pull/3937#issuecomment-513301884) stands if that may help.

**Comment by rjbou** [comment]
there is some [unresolved](https://bugs.busybox.net/show_bug.cgi?id=11981) [issues](https://bugs.busybox.net/show_bug.cgi?id=8826) on busybox [bugtracker](https://bugs.busybox.net/show_bug.cgi?id=6560)...

@talex5 Add `patch` to the package to install is sufficient to make it work (thanks @dbuenzli for the offer ^^). 
In fact, the problem is that `configure` doesn't fail as it finds `patch` command.

And yet another reason [...](https://github.com/ocaml/opam/issues/3782#issuecomment-476177916)

**Comment by rjbou** [comment]
Updated configure!

**Comment by rjbou** [comment]
Fixed by #4034. Thanks!
